### ISSUE TITLE
Slater matrices should be recomputed by default (blocks_between_recompute=1)

### ISSUE THREAD

**Issue by prckent** [QMCPACK/qmcpack#1368]
Currently the full precision build does not recompute the Slater determinants (blocks_between_recompute=0). Eventually this must go wrong, making this a dangerous default. We should have a much safer default (blocks_between_recompute=1). Experts can change this at their leisure.

**Comment by ye-luo** [comment]
This recompute option doesn't call the evaluateLog directly. It is implemented depending on the WFComponent. Currently only SD adds the extra operation of recomputing Orbitials and slaterdet inverse. No recomputing in MSD.

**Comment by prckent** [comment]
Ouch. We will need to implement recompute for MSD as well then. Some of the highest accuracy runs being done are MSD.

**Comment by jtkrogel** [comment]
I wonder if this could be related to the odd behavior @anbenali is seeing for long enough MC trajectories with MSD.  Is it possible that larger MSD would accumulate errors faster in this way?

**Comment by prckent** [comment]
Good point. Maybe. With a long enough run you have a greater chance to be unlucky.

**Comment by ye-luo** [comment]
In the past, not recomputing helped me finding more bugs. The recomputing often hide bugs.
I also didn't get a single solid data point that recomputing is necessary.
If someone claims the recomputing helps the accuracy, I would go for the source of errors(bugs).

For the SD case, both CPU and GPU code do out of place inversion. We probably can do some check before going for the expensive inversion.

**Comment by prckent** [comment]
I have been burned in full production runs in other codes. We will recompute for now. Experts can change the default. Improved algorithms can come later.

**Comment by ye-luo** [comment]
Recomputing does add cost and fortunately it is not heavy.
It should be clear that we do recomputing because this is the only way to stop the error accumulated in the SD inverse matrix. For example, in other parts of the WF, the algorithm of Jastrow is robust against error accumulation and thus no recomputing should be applied.
In MSD, carries a piece of SD for the reference determimant, if its accuracy affects the result, we should secure it with recomputing. If other portions of MSD is sensitive to recomputing, it can potentially be a bug.

**Comment by prckent** [comment]
https://qmcpack.readthedocs.io/en/develop/methods.html#batched-vmc-driver-default

The modern defaults are 1 (mixed precision) and 10 (full/double). While safer, a smarter algorithm that factors electron and step count would be ideal.
